packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -354,6 +354,12 @@ export abstract class BaseService extends Resource
       ...additionalProps
     });
 
+    if (props.deploymentController?.type === DeploymentControllerType.EXTERNAL) {
```

Comment:
Yeah but the problem is if we set it prior, then we need to change the launchType in baseService to be an optional one (otherwise we cannot set it to undefined). This will probably be a breaking change and I did it to avoid any breaking changes. I will submit a new commit addressing on that for reference.

Reply:
I meant above (when you create cfnService and set launchType, you can actually pass in undefined)

Reply:
Perhaps a style preference, but I think it's fine as is!